packages/medusa/src/interfaces/transaction-base-service.ts

```diff
@@ -46,13 +51,17 @@ export abstract class TransactionBaseService {
    * @return the result of the transactional work
    */
   protected async atomicPhase_<TResult, TError>(
-    work: (transactionManager: EntityManager) => Promise<TResult | never>,
+    work: (
+      transactionManager: EntityManager,
+      transactionId?: string
+    ) => Promise<TResult | never>,
     isolationOrErrorHandler?:
       | IsolationLevel
       | ((error: TError) => Promise<never | TResult | void>),
     maybeErrorHandlerOrDontFail?: (
       error: TError
-    ) => Promise<never | TResult | void>
+    ) => Promise<never | TResult | void>,
+    options: Record<string, unknown> & { transactionId?: string } = {}
```

Comment:
suggestion: Should we move the options as the second arguments such as `Options | Something | something` ? to avoid having to pass empty arguments to reach the option?

Reply:
I intentionally made it an additional argument to avoid too many breaking changes, but I am also fine with including it as a second param as it would prob. be used for than others.

Reply:
It should not create any breaking changes by doing that. Let see that later maybe since it will change in a short term with the new transaction management wdyt?